dist/ExtUtils-ParseXS/lib/perlxs.pod

```diff
@@ -1,2065 +1,4803 @@
+        int &i
+      OUTPUT:
+        i
+
+and they both wrap a C function called foo() that takes a single C<int *>
+argument which (presumably) updates the integer pointed to. They both
+generate C code similar to this:
+
+    int i = (int)SvIV(ST(0));
+    foo(&i);
+    sv_setiv(ST(0), (IV)i);
+
+
+=head4 Altering variable initialisation in INPUT
+
+Normally each declared parameter causes a C auto variable of the same name
+to be declared, and for code to be planted which initialises that variable
+to the value of the corresponding passed argument. The initialisation code
+is usually obtained by expanding the typemap template corresponding to
+the parameter's type. It is possible to override, augment, or skip that
+initialisation code, by appending one of the three characters C<= + ;> and
+an initialiser expression, to the C<INPUT> line.
+
+    void
+    foo(a,b,c,d,e,f,g)
+
+        # Use the standard typemap entry:
+        int a
+        # and with optional trailing colon
+        int b;
+
+        # Override the typemap entry:
+        int c = ($type)MySvIV($arg)
+
+        # Skip the initialisation entirely:
+        int d = NO_INIT
+        int e ; NO_INIT
+
+        # Add deferred initialisation code
+        # *in addition* to the standard init:
+        int f + if (some_condition) { $var += 1 }
+
+        # Add deferred initialisation code
+        # *instead of* the standard init:
+        int g ; if (some_condition) { $var += 1 }
+
+Any override code is passed through template expansion in the same way
+that typemap templates are, with C<$var>, C<$arg>, C<$type> etc being
+expanded. Deferred initialisation code is placed after all variable
+declarations.
+
+In modern XS where C<INPUT> is not often used, some of these initialiser
+effects can be achieved in other ways:
+
+=over
+
+=item *
+
+an overridden typemap entry could be specified by using C<TYPEMAP>
+to add a template for the type of this variable;
+
+=item *
+
+skipping initialisation can be achieved using the C<OUT> and C<OUTLIST>
+parameter declaration modifiers;
+
+=item *
+
+adding deferred initialisation code may be achievable via C<PREINIT> or
+C<INIT> blocks.
+
+=back
+
+=head3 The SCOPE: Keyword and typemap entry
+
+    # XSUB-scoped
+    void
+    foo(int i)
+      SCOPE: ENABLE
+      CODE:
+        ...
+
+    # file-scoped
+    SCOPE: ENABLE
+    void
+    bar(int i)
+      CODE:
+        ...
 
-=head2 The Anatomy of an XSUB
+    # typemap entry
+    TYPEMAP: <<EOF
+    INPUT
+    T_MYINT
+       $var = my_int($arg); /* SCOPE */
+    EOF
+
+The C<SCOPE> keyword can be used to enable scoping for a particular XSUB
+(disabled by default). Its effect is to wrap the main body of the XSUB
+(including most parameter and return value processing) within an C<{
+ENTER;> and C<LEAVE; }> pair. This has the effect of clearing any
+accumulated savestack entries at the end of the code body. If disabled,
+then the savestack will usually be cleared by the caller anyway, so this
```

Comment:
"The caller" here means leavesub, right? I don't really understand when one would ever need this. `MULTICALL` shenanigans?